innovation_sweet_spots/analysis/prototyping/public_discourse_analysis/Example_data_preprocessing.py

```diff
@@ -0,0 +1,130 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.3
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Clean text, build sentence corpus and analyse mentions over time
+#
+# - Basic cleaning of articles
+# - Extracting sentences (default context) and noun chunks for subsequent analysis
+# - Analyse mentions of search terms over time
+# - Appendix: alternative options for defining context around search terms
+
+# %% [markdown]
+# ## 1. Import dependencies
+
+# %%
+import os
+import pandas as pd
+import csv
+import spacy
+import pickle
+
+# %%
+# Change first element to location of project folder.
+os.chdir(os.path.join('/Users/jdjumalieva/Documents/Analysis/', 'innovation_sweet_spots'))
+
+# %%
+from innovation_sweet_spots.analysis.prototyping.public_discourse_analysis import pd_data_processing_utils as dpu
+from innovation_sweet_spots.getters.path_utils import OUTPUT_DATA_PATH
+
+# %%
+OUTPUTS_CACHE = OUTPUT_DATA_PATH / ".cache"
+DISC_OUTPUTS_DIR = OUTPUT_DATA_PATH / "discourse_analysis_outputs"
+DISC_OUTPUTS_DIR.mkdir(parents=True, exist_ok=True)
+
+# %%
+search_terms = ['heat pump', 'heat pumps']
+
+# %%
+nlp = spacy.load("en_core_web_sm")
+
+# %% [markdown]
+# ## 2. Clean text, build sentence corpus
+
+# %%
+# Read in article_text dataframe
+article_text = pd.read_csv(os.path.join(DISC_OUTPUTS_DIR, 'article_text_hp.csv'))
+
+# %%
+with open(os.path.join(DISC_OUTPUTS_DIR, 'metadata_dict_hp.pkl'), "rb") as infile:
+        metadata_dict = pickle.load(infile)
+
+# %%
+article_text['year'] = article_text['id'].apply(lambda x: metadata_dict[x]['date'][:4])
```

Comment:
I'm getting an error here - looks like metadata_dict is missing `date` field (NB: I first ran Example_data_collection.py to fetch the data).

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-38-b07bba704d6d> in <module>
----> 1 article_text['year'] = article_text['id'].apply(lambda x: metadata_dict[x]['date'][:4])

~/opt/miniconda3/envs/iss_dev_test/lib/python3.7/site-packages/pandas/core/series.py in apply(self, func, convert_dtype, args, **kwds)
   4136             else:
   4137                 values = self.astype(object)._values
-> 4138                 mapped = lib.map_infer(values, f, convert=convert_dtype)
   4139 
   4140         if len(mapped) and isinstance(mapped[0], Series):

pandas/_libs/lib.pyx in pandas._libs.lib.map_infer()

<ipython-input-38-b07bba704d6d> in <lambda>(x)
----> 1 article_text['year'] = article_text['id'].apply(lambda x: metadata_dict[x]['date'][:4])

KeyError: 'date'
```